jax/interpreters/pxla.py

```diff
@@ -2113,7 +2112,7 @@ def from_hlo(name: str,
       num_replicas, num_partitions = 1, mesh.size
     else:
       num_replicas, num_partitions = mesh.size, 1
-    device_assignment = mesh.device_ids.reshape((num_replicas, num_partitions))
```

Comment:
I wonder if we could even just drop the `device_ids` method from mesh. It seems to be mostly used in the `__repr__` method. (I may be wrong about that.)

Reply:
It looks like there were some usage of `device_ids` in the context of `axis_index_groups`.